drivers/power/max42500/max42500.c

```diff
@@ -0,0 +1,473 @@
+		en0 = NO_OS_GPIO_LOW;
+		en1 = NO_OS_GPIO_LOW;
+		break;
+
+	case MAX42500_STATE_SLEEP:
+		en0 = NO_OS_GPIO_HIGH;
+		en1 = NO_OS_GPIO_LOW;
+		break;
+
+	case MAX42500_STATE_ON:
+		en0 = NO_OS_GPIO_HIGH;
+		en1 = NO_OS_GPIO_HIGH;
+		break;
+
+	default:
+		en0 = NO_OS_GPIO_HIGH;
+		en1 = NO_OS_GPIO_HIGH;
+		break;
+	}
+
+	ret = no_os_gpio_set_value(desc->en0, en0);
+	if (ret)
+		return ret;
+
+	return no_os_gpio_set_value(desc->en1, en1);
+}
+
+/**
+ * @brief Read a raw value from a register.
+ * @param desc - The device structure.
+ * @param reg - Register structure holding info about the register to read from.
+ * @return 0 in case of success, error code otherwise
+ */
+int max42500_reg_read(struct max42500_dev *desc, struct max42500_reg_st *reg)
+{
+	int ret;
+	uint8_t i2c_data[MAX42500_I2C_RD_FRAME_SIZE] = {0};
+	uint8_t bytes_number;
+	uint8_t crc;
+
+	/* PEC is computed over entire I2C frame from the first START condition */
+	i2c_data[0] = (desc->comm_desc->slave_address << 1);
+	i2c_data[1] = reg->addr;
+	i2c_data[2] = (desc->comm_desc->slave_address << 1) | 0x1;
+
+	/* I2C write target address */
+	bytes_number = 1;
+
+	ret = no_os_i2c_write(desc->comm_desc, &i2c_data[1], bytes_number, 0);
+	if (ret)
+		return ret;
+
+	/* Change read byte count if PECE is enabled (1-byte data. 1-byte PEC) */
+	bytes_number = (desc->pece) ? 2 : bytes_number;
+
+	ret = no_os_i2c_read(desc->comm_desc, &i2c_data[3], bytes_number, 1);
+	if (ret)
+		return ret;
+
+	if (desc->pece) {
+		/* Compute CRC over entire I2C frame */
+		crc = no_os_crc8(max42500_crc8, i2c_data,
+				 (MAX42500_I2C_RD_FRAME_SIZE - 1), 0);
+
+		if (i2c_data[4] != crc)
+			return -ECOMM;
+	}
+
+	/* Update register with new read value */
+	reg->value = i2c_data[3];
+
+	return 0;
+}
+
+/**
+ * @brief Write a raw value to a register.
+ * @param desc - The device structure.
+ * @param reg - Register structure holding info about the register to be written.
+ * @return 0 in case of success, error code otherwise
+ */
+int max42500_reg_write(struct max42500_dev *desc, struct max42500_reg_st reg)
+{
+	uint8_t i2c_data[MAX42500_I2C_WR_FRAME_SIZE] = {0};
+	uint8_t bytes_number;
+
+	bytes_number = (desc->pece) ? (MAX42500_I2C_WR_FRAME_SIZE - 1) : 2;
+	i2c_data[0] = (desc->comm_desc->slave_address << 1);
+	i2c_data[1] = reg.addr;
+	i2c_data[2] = reg.value;
+
+	if (desc->pece)
+		i2c_data[3] = no_os_crc8(max42500_crc8, i2c_data,
+					 bytes_number, 0);
+
+	return no_os_i2c_write(desc->comm_desc, &i2c_data[1], bytes_number, 1);
+}
+
+/**
+ * @brief Wrap the write register function to give it a modern signature.
+ * @param desc - The device structure
+ * @param reg - Address of the register to be written.
+ * @param writeval - New value for the register.
+ * @return 0 in case of success, error code otherwise.
+ */
+int max42500_reg_write2(struct max42500_dev *desc,
+			uint32_t reg,
+			uint32_t writeval)
+{
+	desc->regs[reg].value = writeval;
+
+	return max42500_reg_write(desc, desc->regs[reg]);
+}
```

Comment:
There is no need to have 2 of these "register write" functions. In order to keep compatibility with the rest of the no os drivers, you should only implement this one.

Reply:
This is noted.